scalableminds/webknossos
# Refactor TaskType Resolution Restrictions

The old format for tasktype settings
```
  +allowedMagnifications?: {
    shouldRestrict: boolean,
    min?: number,
    max?: number,
  },
```
had threefold redundancy (field is optional, shouldRestrict boolean, numbers are optional), creating a lot of possibilities for inconsistent states. I propose we change that to 
```
  +resolutionRestrictions: {
    min?: number,
    max?: number,
  },
```
The old format fitted the form in tasktype creat view better, though, there needs to be an adapter for the formValues.

### TODO:
- [x] update schema + scala code
- [x] sql evolution
- [x] use new format in task type create view
- [x] use new format in flycam accessor

### Steps to test:
- create task type with restrictions, create task
- fetch task, resolutions should be restricted

### Issues:
- fixes #4901 

------
